File: app/src/main/java/im/tox/antox/callbacks/AntoxOnNameChangeCallback.java
package im.tox.antox.callbacks;

import android.content.Context;
import android.content.Intent;
import android.support.v4.content.LocalBroadcastManager;

import im.tox.antox.data.AntoxDB;
import im.tox.antox.tox.ToxSingleton;
import im.tox.antox.utils.AntoxFriend;
import im.tox.antox.utils.Constants;
import im.tox.jtoxcore.callbacks.OnNameChangeCallback;

public class AntoxOnNameChangeCallback implements OnNameChangeCallback<AntoxFriend> {

    private final static String TAG = "im.tox.antox.TAG";
    private Context ctx;

    ToxSingleton toxSingleton = ToxSingleton.getInstance();
    public AntoxOnNameChangeCallback(Context ctx) {
        this.ctx = ctx;
    }

    @Override
    public void execute(AntoxFriend friend, String newName) {
        AntoxDB db = new AntoxDB(ctx);
        db.updateFriendName(friend.getId(), newName);
        db.close();
        toxSingleton.updateFriendsList(ctx);
    }
}

File: app/src/main/scala/im/tox/antox/callbacks/AntoxOnNameChangeCallback.scala
package im.tox.antox.callbacks

import android.content.Context
import im.tox.antox.data.AntoxDB
import im.tox.antox.tox.ToxSingleton
import im.tox.antox.utils.UIUtils
import im.tox.tox4j.core.callbacks.FriendNameCallback

object AntoxOnNameChangeCallback {

  private val TAG = "im.tox.antox.TAG"
}

class AntoxOnNameChangeCallback(private var ctx: Context) extends FriendNameCallback {
  override def friendName(friendNumber: Int, nameBytes: Array[Byte]): Unit = {
    val name = UIUtils.removeNewlines(new String(nameBytes, "UTF-8"))
    ToxSingleton.getAntoxFriend(friendNumber) match {
      case Some(friend) => friend.setName(name)
      case None => throw new Exception("Friend not found.")
    }

    val db = new AntoxDB(ctx)
    db.updateContactName(ToxSingleton.getAntoxFriend(friendNumber).get.getKey, name)
    db.close()
    ToxSingleton.updateFriendsList(ctx)
  }
}